crypto-ffi/bindings/swift/WireCoreCrypto/WireCoreCrypto/Types.swift

```diff
@@ -0,0 +1,698 @@
+    // declare one manually.
+    public init(message: Data?, isActive: Bool, commitDelay: UInt64?, senderClientId: ClientId?, hasEpochChanged: Bool, identity: WireIdentity, crlNewDistributionPoints: [String]?) {
+        self.message = message
+        self.isActive = isActive
+        self.commitDelay = commitDelay
+        self.senderClientId = senderClientId
+        self.hasEpochChanged = hasEpochChanged
+        self.identity = identity
+        self.crlNewDistributionPoints = crlNewDistributionPoints
+    }
+}
+
+extension WireCoreCryptoUniffi.BufferedDecryptedMessage {
+    
+    func lift() -> BufferedDecryptedMessage {
+        BufferedDecryptedMessage(
+            message: message,
+            isActive: isActive,
+            commitDelay: commitDelay,
+            senderClientId: senderClientId,
+            hasEpochChanged: hasEpochChanged,
+            identity: identity.lift(),
+            crlNewDistributionPoints: crlNewDistributionPoints
+        )
+    }
+    
+}
+
+public struct DecryptedMessage {
+    public var message: Data?
+    public var isActive: Bool
+    public var commitDelay: UInt64?
+    public var senderClientId: ClientId?
+    public var hasEpochChanged: Bool
+    public var identity: WireIdentity
+    public var bufferedMessages: [BufferedDecryptedMessage]?
+    public var crlNewDistributionPoints: [String]?
+
+    // Default memberwise initializers are never public by default, so we
+    // declare one manually.
+    public init(message: Data?, isActive: Bool, commitDelay: UInt64?, senderClientId: ClientId?, hasEpochChanged: Bool, identity: WireIdentity, bufferedMessages: [BufferedDecryptedMessage]?, crlNewDistributionPoints: [String]?) {
+        self.message = message
+        self.isActive = isActive
+        self.commitDelay = commitDelay
+        self.senderClientId = senderClientId
+        self.hasEpochChanged = hasEpochChanged
+        self.identity = identity
+        self.bufferedMessages = bufferedMessages
+        self.crlNewDistributionPoints = crlNewDistributionPoints
+    }
+}
+
+extension WireCoreCryptoUniffi.DecryptedMessage {
+    
+    func lift() -> DecryptedMessage {
+        DecryptedMessage(
+            message: message,
+            isActive: isActive,
+            commitDelay: commitDelay,
+            senderClientId: senderClientId,
+            hasEpochChanged: hasEpochChanged,
+            identity: identity.lift(),
+            bufferedMessages: bufferedMessages?.map { $0.lift() },
+            crlNewDistributionPoints: crlNewDistributionPoints
+        )
+    }
+    
+}
+
+public enum MlsWirePolicy : UInt8 {
+    
+    /**
+     * Handshake messages are never encrypted
+     */
+    case plaintext = 1
+    /**
+     * Handshake messages are always encrypted
+     */
+    case ciphertext = 2
+    
+    func lower() -> WireCoreCryptoUniffi.MlsWirePolicy {
+        switch self {
+        case .plaintext: .plaintext
+        case .ciphertext: .ciphertext
+        }
+    }
+}
```

Comment:
There are quite a few of cases like this. What is the reason we can't have a type alias to the internal type?
Missing docs?

The problem I see with this approach is that we're duplicating definitions (1 on the Rust side, 1 here).
With Wasm, some effort was done to deduplicate types, and the idea is to continue that way, i.e. have only a single type definition. So I would like us to find a better solution here.

Reply:
There's no other way if we don't want expose the `WireCoreCryptoUniffi` module. We have exactly the same problem and solution in the Kotlin bindings.